core/domain/DomainBase.php

```diff
@@ -128,35 +135,42 @@ public function versionValueObject()
     /**
      * @return string
      */
-    public function sourceAssetsPath()
+    public function distributionAssetsPath()
     {
-        return $this->pluginPath() . 'assets/src/';
+        return $this->pluginPath() . 'assets/dist/';
     }
 
 
     /**
      * @return string
      */
-    public function sourceAssetsUrl()
+    public function distributionAssetsUrl()
     {
-        return $this->pluginUrl() . 'assets/src/';
+        return $this->pluginUrl() . 'assets/dist/';
     }
 
 
     /**
      * @return string
      */
-    public function distributionAssetsPath()
+    public function assetNamespace()
     {
-        return $this->pluginPath() . 'assets/dist/';
+        return $this->asset_namespace;
     }
 
 
     /**
-     * @return string
+     * @return void
      */
-    public function distributionAssetsUrl()
+    private function setAssetNamespace()
     {
-        return $this->pluginUrl() . 'assets/dist/';
+        $asset_namespace  = explode('/', $this->plugin_basename);
+        $asset_namespace = reset($asset_namespace);
+        $asset_namespace = $asset_namespace === 'event-espresso-core'
+            ? Registry::ASSET_NAMESPACE_CORE
```

Comment:
Would this maybe be better as:

```php
$asset_namespace = $asset_namespace === Registry::ASSET_NAMESPACE_CORE
    ? Registry::ASSET_NAMESPACE_CORE
    : $asset_namespace
```

Reply:
no, we are actually changing 'event-espresso-core' into 'core' because

```php
const ASSET_NAMESPACE_CORE     = 'core';
```

because 'event-espresso-core' is coming from the plugin basename...
oooooo... i just realized this is not going to work because if someone renames their EE folder, 
then the first part of their plugin basename will NOT be 'event-espresso-core' !!!
dang it.. going to hit you up in chat to discuss this...